<launch>
    <!-- load params -->
    <rosparam file="$(find dataset_builder)/params/dicts.yaml" command="load"/>

    <!-- get input args -->
    <arg name="robofleet" default='false'/>
    <arg name="outdir" default="$(find dataset_builder)/dataset"/>
    <arg name="class" default="Advance"/>
    <arg name="bagdata" default='true'/>
    <arg name="image_fps" default="20"/>

    <!-- configure topics -->
    <arg name="start_topic_name"  default="/hololens/natural_input_capture/start"/>
    <arg name="stop_topic_name"   default="/hololens/natural_input_capture/end"/>
    <arg name="mono_ll_topic"     default="/hololens/LEFT_LEFT/image_raw"/>
    <arg name="mono_lf_topic"     default="/hololens/LEFT_FRONT/image_raw"/>
    <arg name="mono_rf_topic"     default="/hololens/RIGHT_FRONT/image_raw"/>
    <arg name="mono_rr_topic"     default="/hololens/RIGHT_RIGHT/image_raw"/>
    <arg name="mic_topic"         default="/hololens/microphone/data"/>

    <!-- set params -->
    <param name="class"    value="$(arg class)"/>
    <param name="base_outdir"   value="$(arg outdir)"/>
    <param name="timestamp"     value=""/>
    <param name="class_name_id" value=""/>
    
    <!-- robofleet client -->
    <group if="$(arg robofleet)" ns="robofleet"> 
        <arg name="rfconfig" default="$(env HRI_CACTI_XR_PATH)/robofleet/hl-ss-config.yaml"/>
	    <node
	    	name    = "robofleet_client"
	    	pkg     = "robofleet_client"
	    	type    = "client"
	    	output  = "screen"
	    	args	= "$(arg rfconfig)"
	    />
    </group>

    <!-- config params -->
    <node ns="config_params_node" name = "config_params" pkg = "dataset_builder" type = "config_params.py" output = "screen"/>

    <!-- mono left left camera stream -->
    <group ns="mono_ll">
        <include file="$(find image_writer)/launch/save_raw_image_stream.launch" />
        <param name ="filepath" value="$(arg outdir)/devices/mono_ll/raw/$(arg class)/"/>
        <param name ="topicname_image" value="$(arg mono_ll_topic)"/>
        <param name ="fps" value="$(arg image_fps)"/>
    </group>



</launch>